src/Http/Controllers/SessionController.php

```diff
@@ -58,6 +75,30 @@ protected function getAllowedUsers($params) {
         return $usersAllowed;
     }
 
+    protected function checkGoogleAuthAndGetUser($renderingId, $accessToken) {
+        $client = new Client();
+        $path = '/renderings/'.$renderingId.'/google-authorization';
+        $options = array(
+            'headers' => array(
+                'Content-Type' => 'application/json',
+                'forest-secret-key' => Config::get('forest.secret_key'),
+                'google-access-token' => $accessToken,
+            )
+        );
+        $response = $client->request('GET', $path, $options);
+        $response = json_decode($response->getBody());
```

Comment:
You could also easily factorize the beginning of the method here as it is the same logic you have in `getAllowedUsers`.